src/frontend/src/components/public/PIAFormTabs/Next_Steps/index.tsx

```diff
@@ -26,24 +26,28 @@ export const PIANextSteps = () => {
   };
 
   useEffect(() => {
-    if (pia.hasAddedPiToDataElements === false) {
-      if (pia.isNextStepsSeenForDelegatedFlow) {
-        // redirect to view page
-        navigateFn(routes.PIA_VIEW);
+    // Delay the execution of the code inside setTimeout to ensure it runs after the component is initially rendered
+    setTimeout(() => {
+      if (pia.hasAddedPiToDataElements === false) {
+        if (pia.isNextStepsSeenForDelegatedFlow) {
+          // redirect to view page
+          navigateFn(routes.PIA_VIEW);
+        } else {
+          piaStateChangeHandler(true, 'isNextStepsSeenForDelegatedFlow');
+          piaStateChangeHandler(false, 'isNextStepsSeenForNonDelegatedFlow');
+        }
       } else {
-        piaStateChangeHandler(true, 'isNextStepsSeenForDelegatedFlow');
-        piaStateChangeHandler(false, 'isNextStepsSeenForNonDelegatedFlow');
+        // if true or null
+        if (pia.isNextStepsSeenForNonDelegatedFlow) {
+          // redirect to next tab
+          navigateFn(routes.PIA_DISCLOSURE_EDIT);
+        } else {
+          // Update the state variables based on conditions.
+          piaStateChangeHandler(true, 'isNextStepsSeenForNonDelegatedFlow');
+          piaStateChangeHandler(false, 'isNextStepsSeenForDelegatedFlow');
+        }
       }
-    } else {
-      // if true or null
-      if (pia.isNextStepsSeenForNonDelegatedFlow) {
-        // redirect to next tab
-        navigateFn(routes.PIA_DISCLOSURE_EDIT);
-      } else {
-        piaStateChangeHandler(true, 'isNextStepsSeenForNonDelegatedFlow');
-        piaStateChangeHandler(false, 'isNextStepsSeenForDelegatedFlow');
-      }
-    }
+    }, 1);
```

Comment:
Can you not just add `pia.isNextStepsSeenForDelegatedFlow` to the useEffect dependency array?

Reply:
> Can you not just add `pia.isNextStepsSeenForDelegatedFlow` to the useEffect dependency array?

`useEffect(() => {
  if (pia.hasAddedPiToDataElements === false) {
    code....
  }
}, [pia.isNextStepsSeenForDelegatedFlow]);`      
do you mean like this? I have tried but it didn't work..